c7n/resolver.py

```diff
@@ -141,5 +142,6 @@ def get_values(self):
                 return jmespath.search(self.data['expr'], data)
             return data
         elif format == 'txt':
+            contents = contents.encode('utf8')
             return [s.strip().decode('utf8')
                     for s in io.BytesIO(contents).readlines()]
```

Comment:
Why not just use a `StringIO` as w/ the `csv{2dict}` case? `get_contents` returns a `six.text_type`, and it seems that's what we want to return here, as well. Isn't the `decode`/`encode` dance make-work?

Reply:
I’m not sure, but it was intentionally changed in #1450. Without any information on why, I didn’t want to change it back.